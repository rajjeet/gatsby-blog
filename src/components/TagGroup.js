import React from 'react';
import {navigate} from "gatsby";
import styled from 'styled-components';
import * as theme from '../utils/colors'

const TagGroup = ({className, tags, getSlug}) => (
    <div className={className}>
        {
            tags &&
            tags.map(tag =>
                    <span key={tag.fieldValue}
                          onClick={() => navigate(getSlug(tag.fieldValue) + '1')}>{tag.fieldValue}
                        {tag.totalCount && <span>{tag.totalCount}</span>}
            </span>
            )
        }
    </div>
);

const StyledTagGroup = styled(TagGroup)`
  display: ${props => props.inline ? 'inline' : 'inline-block'};
  
  > span {
    padding: .2em .55em;
    margin-right: .5em;
    margin-bottom: .5em;
    box-shadow: ${theme.lightbBoxShadow};
    display: inline-block;
    border-radius: 5px;
    color: white;
    font-weight: bolder;    
    font-size: .8em;
    background-color: ${theme.primaryColor};
    :hover {
      box-shadow: ${theme.hoverBoxShadow};
      transform: translateY(-2px);
      transition: ease .3s;
      cursor: pointer;
    }
    > span {
      margin-left: 1em;
    };    
  }
  
`;

export default StyledTagGroup;