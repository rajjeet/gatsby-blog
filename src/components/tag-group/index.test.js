import React from 'react';
import renderer from 'react-test-renderer';
import TagGroup from './index';

const makeProps = () => ({
  className: {}, tags: [], getSlug: jest.fn(),
});

describe('<TagGroup />', () => {
  it('should render', () => {
    renderer.create(<TagGroup {...makeProps()} />);
  });
});
