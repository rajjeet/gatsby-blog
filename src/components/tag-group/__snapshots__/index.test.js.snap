// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TagGroup /> should render 1`] = `
<DocumentFragment>
  <div
    class="[object Object] sc-bdVaJa fdxsEr"
  >
    <span
      role="link"
      tabindex="0"
    >
      Experience
      <span>
        1
      </span>
    </span>
    <span
      role="link"
      tabindex="0"
    >
      Guide
      <span>
        2
      </span>
    </span>
  </div>
</DocumentFragment>
`;
